### PR TITLE
Add TransportStatus tracker

### PR DESCRIPTION
**Pull request by rudigiesler**
We need a tool to make sure that we don't send the same TransportStatus repeatedly, we only want to send a status event when the status changes.

### CODE REVIEW

File: vumi/utils.py [hunk 1]
```diff
@@ -452,3 +452,54 @@ def safe_routing_key(routing_key):
...
+
+    def _check_state(self, status, component):
+        state = self._get_state(component)
+        if state != status:
+            self._remove_types(component)
+            self._set_state(component, status)
+
+    def _check_type(self, type, component):
+        types = self._get_types(component)
+        if type not in types:
```
Comment (justinvdm) [inline vumi/utils.py]: A little bikesheddy, but could we avoid the `type` reserved word?

File: vumi/utils.py [hunk 2]
```diff
@@ -452,3 +452,54 @@ def safe_routing_key(routing_key):
...
+            self._add_type(component, type)
+            return True
```
Comment (justinvdm) [inline vumi/utils.py]: Should we maybe return `False` otherwise?

**Comment by justinvdm** [comment]
Left two minor comments, looks good.

**Comment by rudigiesler** [comment]
Comments addressed

File: vumi/utils.py [hunk 3]
```diff
@@ -452,3 +452,55 @@ def safe_routing_key(routing_key):
...
+    def _set_state(self, component, state):
+        self.state[component] = state
+
+    def _get_types(self, component):
+        return self.types.get(component, set())
+
+    def _add_type(self, component, type):
+        if component not in self.types:
+            self.types[component] = set()
+        self.types[component].add(type)
```
Comment (justinvdm) [inline vumi/utils.py]: Could we avoid using `type` here as well?

**Comment by justinvdm** [comment]
:+1:

**Pull request merged**